sdk/resource/internal/schema/resconv/convert.go

```diff
@@ -0,0 +1,134 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resconv // import "go.opentelemetry.io/otel/sdk/resource/internal/schema/resconv"
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/Masterminds/semver/v3"
+
+	"go.opentelemetry.io/otel/attribute"
+	ast10 "go.opentelemetry.io/otel/schema/v1.0/ast"
+	"go.opentelemetry.io/otel/schema/v1.1/ast"
+	"go.opentelemetry.io/otel/schema/v1.1/types"
+)
+
+// Upgrade upgrades attrs in place with schema.
+func Upgrade(schema *ast.Schema, attrs []attribute.KeyValue) error {
```

Comment:
I think this will need the schema it's coming from and the schema it's going to.  Otherwise, this is going to apply all upgrades for the destination schema.

I built a few tests to see if this is working, [here](https://github.com/MrAlias/opentelemetry-go/pull/873).  Let me know if there is a better way for you to get these tests and consume them.

Reply:
0be6881acc808c11b18d52b1c1a0bf3fcc696207